cpu/stm32_common/periph/flash_common.c

```diff
@@ -38,15 +38,17 @@
 
 void _unlock(void)
 {
-    DEBUG("[flash-common] unlocking the flash module\n");
     if (CNTRL_REG & CNTRL_REG_LOCK) {
+        DEBUG("[flash-common] unlocking the flash module\n");
         KEY_REG = FLASH_KEY1;
         KEY_REG = FLASH_KEY2;
     }
 }
 
 void _lock(void)
 {
-    DEBUG("[flash-common] locking the flash module\n");
-    CNTRL_REG |= CNTRL_REG_LOCK;
+    if (!(CNTRL_REG & CNTRL_REG_LOCK)) {
```

Comment:
Is there still a case where this is necessary ?
But maybe this would also be needed to ask it for `_unlock`.

Reply:
At least 'test 255' and 'write_raw 0x807f808 abcdefgh' work on M3 nodes without both protections.

But there may be good reasons for it just asking.

Reply:
There is the same check in the _unlock function, I wanted it to be consistent. The idea is "if already locked, do nothing".